docs/pages/docs/getting-started.mdx

```diff
@@ -73,43 +125,67 @@ If your git repo's base branch is NOT `origin/master` then you need to specify a
+Packages that do not have have this script defined in their `package.json`'s list of `scripts` will be ignored and skipped over during `turbo run`.
 
-```jsonc
+```jsonc filename="./turbo.json"
 {
   "$schema": "https://turborepo.org/schema.json",
-  "baseBranch": "origin/main",
   "pipeline": {
     "build": {
+      // A package's `build` script depends on that package's
+      // topological dependencies' and devDependencies'
+      // `build` tasks  being completed first
+      // (that's what the `^` symbol signifies).
       "dependsOn": ["^build"],
       // note: output globs are relative to each package's `package.json`
       // (and not the monorepo root)
       "outputs": [".next/**"]
     },
     "test": {
-      "dependsOn": ["^build"],
-      "outputs": []
+      // A package's `test` script depends on that package's
+      // own `build` script being completed first.
+      "dependsOn": ["build"],
+      "outputs": [],
+      // A package's `test` script should only be rerun when
+      // either a `.tsx` or `.ts` file has changed in `src` or `test` folders.
+      "inputs": ["src/**/*.tsx", "src/**/*.ts", "test/**/*.ts", "test/**/*.tsx"]
     },
     "lint": {
+      // A package's `lint` script has no dependencies and
+      // can be run whenever. It also has no filesystem outputs.
       "outputs": []
     },
-    "dev": {
-      "cache": false
+    "deploy": {
+      // A package's `deploy` script depends on the `build`,
+      // `test`, and `lint` scripts of the same package
+      // being completed. It also has no filesystem outputs.
+      "dependsOn": ["build", "test", "lint"],
+      "outputs": []
     }
   }
 }
 ```
 
-In the above example, the `build` and `test` tasks are dependent on their packages `dependencies` and `devDependencies` being built first, this is denoted with the `^` prefix.
+The rough execution order for a given package based on the `dependsOn` keys above will be:
+
+1. `build` once its dependencies have run their `build` commands
+2. `test` once its _own_ `build` command is finished and has no filesystem outputs (just logs)
+3. `lint` can run whenever and has no filesystem outputs (just logs)
+4. `deploy` once its _own_ `build`, `test`, and `lint` commands have finished, no outputs
+
+The full pipeline can then be run:
 
-For each script in each workspace's `package.json`, Turborepo will cache files outputted to `dist/**` and `build/**` folders relative to each packages's `package.json` by default if an override isn't added. Using the [`outputs`](./features/caching#configuring-cache-outputs-1) array allows you to override the default cache folders,
-like in the example above, where the `.next/**` folder is selected to be the default cache outputs folder for the `build` task for each package. Turborepo will automatically record and cache logs to `.turbo/turbo-<script>.log` for you, so you don't ever need to specify that in the `outputs` array.
+```bash
+npx turbo run build test lint deploy
```

Comment:
We should not be recommending that people run this via `npx`, that's a download of `turbo`, executing a script, a download of `turbo-windows-arm64`, and then finally execution.

We should add a separate section about doing installation.

Reply:
@nathanhammond if you want to contribute that section to this PR please feel free to! 😄